utilsForFrontend.mjs

```diff
@@ -468,8 +468,8 @@ export function isVisible(fnodeOrElement) {
         if (style.visibility === 'hidden' ||
             style.display === 'none' ||
             style.opacity === '0' ||
-            style.width === '0' ||
-            style.height === '0') {
+            style.width === '0px' ||
```

Comment:
While we're Doing Right Things, we might as well also cover cases like "0%", which I just happened to hit while testing this. My gosh, there are going to be a lot of them. :-( Maybe we should just rely on getBoundingClientRect instead? Would that always show a zero height or width when there is one?

Reply:
Ah, okay, what if we use a variation on Emilio's recommended version from the [Price Tracker issue](https://github.com/mozilla/price-tracker/issues/319) which omits `getComputedStyle().width/height` entirely, but add the `else` statement you have here regarding offscreen elements?

Reply:
Sounds good. I assume we'd add something to maintain this offscreen-detection functionality from the current version:

```js
            // It wasn't hidden based on a computed style. See if it's
            // offscreen:
            const rect = element.getBoundingClientRect();
            const frame = element.ownerDocument.defaultView;  // window or iframe
            if ((rect.right + frame.scrollX < 0) ||
                (rect.bottom + frame.scrollY < 0)) {
                return false;
            }
```

Reply:
The updated implementation should cover that scenario, as we `window.scrollTo` to update the viewport location before checking for clickability.

I actually ended up going with the third option checking for clickability (see "Updating isVisible: second attempt" in [this comment](https://github.com/mozilla/price-tracker/issues/319#issue-468970781)), but added the workaround that doesn't require privileged Firefox code. This yields a 53% reduction in Fathom-related jank (see the commit message for more details and actual profiles).

If/when the CSS working group updates `element(s)FromPoint` to include the option to ignore viewport clipping, then we can modify this method and we'll see the 81% reduction as described in that comment.